front/components/actions/retrieval/utils.ts

```diff
@@ -1,26 +1,17 @@
-import type { Citation } from "@dust-tt/sparkle";
 import type { RetrievalDocumentType } from "@dust-tt/types";
 import {
   getProviderFromRetrievedDocument,
   getTitleFromRetrievedDocument,
 } from "@dust-tt/types";
 
-interface RetrievedDocumentCitation {
-  href?: string;
-  title: string;
-  type: Exclude<React.ComponentProps<typeof Citation>["type"], undefined>;
+export function makeDocumentCitation(document: RetrievalDocumentType) {
```

Comment:
Can we keep the return type on those functions? It seems pretty easy to forget/break something.